gk/fib.c

```diff
@@ -1487,33 +1465,75 @@ add_fib_entry(const char *prefix, const char *gt_ip, const char *gw_ip,
 	 * accidentally create a security hole.
 	 */
 	rte_spinlock_lock_tm(&gk_conf->lpm_tbl.lock);
-	ret = check_prefix_locked(&prefix_info, action, gk_conf);
+	ret = check_prefix_locked(prefix_info, action, gk_conf);
 	if (ret < 0) {
 		rte_spinlock_unlock_tm(&gk_conf->lpm_tbl.lock);
 		return -1;
 	}
 
 	ret = add_fib_entry_locked(
-		&prefix_info, gt_para, gw_para, action, gk_conf);
+		prefix_info, gt_addr, gw_addr, action, gk_conf);
 	rte_spinlock_unlock_tm(&gk_conf->lpm_tbl.lock);
 
 	return ret;
 }
 
 int
-del_fib_entry(const char *ip_prefix, struct gk_config *gk_conf)
+add_fib_entry(const char *prefix, const char *gt_ip, const char *gw_ip,
+	enum gk_fib_action action, struct gk_config *gk_conf)
 {
 	int ret;
 	struct ip_prefix prefix_info;
+	struct ipaddr gt_addr, gw_addr;
+	struct ipaddr *gt_para = NULL, *gw_para = NULL;
 
-	prefix_info.str = ip_prefix;
-	prefix_info.len = parse_ip_prefix(ip_prefix, &prefix_info.addr);
+	prefix_info.str = prefix;
+	prefix_info.len = parse_ip_prefix(prefix, &prefix_info.addr);
 	if (prefix_info.len < 0)
 		return -1;
 
+	if (gt_ip != NULL) {
+		ret = convert_str_to_ip(gt_ip, &gt_addr);
+		if (ret < 0)
+			return -1;
+		gt_para = &gt_addr;
+	}
+
+	if (gw_ip != NULL) {
+		ret = convert_str_to_ip(gw_ip, &gw_addr);
+		if (ret < 0)
+			return -1;
+		gw_para = &gw_addr;
+	}
+
+	return add_fib_entry_numerical(&prefix_info,
+		gt_para, gw_para, action, gk_conf);
+}
+
+int
+del_fib_entry_numerical(
+	struct ip_prefix *prefix_info, struct gk_config *gk_conf)
+{
+	int ret;
+
+	if (prefix_info == NULL || prefix_info->str == NULL
+			|| prefix_info->len < 0)
+		return -1;
```

Comment:
Given that `prefix_info` is a required parameter, we don't need the extra tests `prefix_info == NULL || prefix_info->str == NULL` because if the caller forgets them, the code is going to halt. The extra test would make sense if `del_fib_entry_numerical()` could be called from Lua, but there's no reason to call `del_fib_entry_numerical()` from Lua.